src/swap/types.ts

```diff
@@ -3,47 +3,82 @@ import { Token } from '../token/types';
 
 export type AddressOrETH = Address | 'ETH';
 
+/**
+ * Note: exported as public Type
+ */
 export type Fee = {
-  amount: string;
-  baseAsset: Token;
-  percentage: string;
+  amount: string; // The amount of the fee
+  baseAsset: Token; // The base asset for the fee
+  percentage: string; // The percentage of the fee
 };
 
+export type GetQuoteAPIParams = {
+  from: AddressOrETH | ''; // The source address or 'ETH' for Ethereum
+  to: AddressOrETH | ''; // The destination address or 'ETH' for Ethereum
+  amount: string; // The amount to be swapped
+  amountReference?: string; // The reference amount for the swap
+};
+
+/**
+ * Note: exported as public Type
+ */
+export type GetQuoteParams = {
+  from: Token; // The source token for the swap
+  to: Token; // The destination token for the swap
+  amount: string; // The amount to be swapped
+  amountReference?: string; // The reference amount for the swap
+  amountInDecimals?: boolean; // Whether the amount is in decimals
+};
+
+/**
+ * Note: exported as public Type
+ */
+export type GetQuoteResponse = Quote | SwapError;
+
+/**
+ * Note: exported as public Type
+ */
 export type Quote = {
-  amountReference: string;
-  fromAmount: string;
-  fromAsset: Token;
-  highPriceImpact: boolean;
-  priceImpact: string;
-  slippage: string;
-  toAmount: string;
-  toAsset: Token;
-  warning?: QuoteWarning;
+  amountReference: string; // The reference amount for the quote
+  from: Token; // The source token for the swap
+  fromAmount: string; // The amount of the source token
+  highPriceImpact: boolean; // Whether the price impact is high
```

Comment:
`highPriceImpact` sounds odd as `boolean`. I will expect this to have some prefix like `is` or `has`.

cc @0xAlec @cpcramer @kyhyco

Reply:
I think this makes sense to me and is relatively intuitive, but not against adding a prefix either

```
const swap = await getQuote(params...)

if (swap.highPriceImpact) {
// display a warning to the user
}
```